optuna/samplers/_cmaes.py

```diff
@@ -484,6 +519,26 @@ def _init_optimizer(
                 population_size=population_size,
             )
 
+        if self._with_margin:
+            steps = []
+            for dist in trans._search_space.values():
+                assert isinstance(dist, (IntDistribution, FloatDistribution))
+                # Set step 0.0 for continuous search space.
+                steps.append(0.0 if dist.step is None else dist.step)
+
+            # If there is no discrete search space, we use `CMA` because CMAwM` throws an error.
+            if any(step > 0.0 for step in steps):
+                return CMAwM(
+                    mean=mean,
+                    sigma=sigma0,
+                    bounds=trans.bounds,
+                    steps=np.array(steps, dtype=float),
+                    cov=cov,
+                    seed=self._cma_rng.randint(1, 2**31 - 2),
+                    n_max_resampling=10 * n_dimension,
+                    population_size=population_size,
+                )
```

Comment:
nit: Perhaps either would be fine, but left one minor suggestion.

```suggestion
            steps = np.empty(len(trans._search_space), dtype=float)
            for i, dist in enumerate(trans._search_space.values()):
                assert isinstance(dist, (IntDistribution, FloatDistribution))
                # Set step 0.0 for continuous search space.
                steps[i] = dist.step or 0.0

            # If there is no discrete search space, we use `CMA` because CMAwM` throws an error.
            if np.any(steps > 0.0):
                return CMAwM(
                    mean=mean,
                    sigma=sigma0,
                    bounds=trans.bounds,
                    steps=steps,
                    cov=cov,
                    seed=self._cma_rng.randint(1, 2**31 - 2),
                    n_max_resampling=10 * n_dimension,
                    population_size=population_size,
                )
```